code/game/objects/effects/temporary_visuials/temproary_visual.dm

```diff
@@ -0,0 +1,36 @@
+//temporary visual effects
+/obj/effect/temp_visual
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "nothing"
+	anchored = TRUE
+	layer = ABOVE_MOB_LAYER
+	mouse_opacity = 0
+	var/duration = 10 //in deciseconds
+	var/randomdir = TRUE
+
+/obj/effect/temp_visual/initialize()
+	//. = ..()
```

Comment:
Sorry, didn't notice this before, but initialize needs to call its parent and return an init hint.  (easiest way to do this is to do . = ..()

And that TG code looks nice to me, so I recommend adding it back in.
```
	. = ..()
	if(randomdir)
		set_dir(pick(global.cardinal))
```